### PR TITLE
	Remove indirection in checking propTypes

### PR DESCRIPTION
**Pull request by gaearon**
*TLDR: I’m removing `react/lib/checkPropTypes` import in favor of `React.checkPropTypes` public API access from renderer, and direct `checkPropTypes` access from the core.*

Shared `checkReactTypeSpec` currently depends on `react/lib/checkPropTypes` which is gone in flat bundles. So this breaks in React Native sync process. In DOM, we currently ship two copies (in both React and ReactDOM flat bundles) but this breaks until RN is also a flat bundle. I could shim it but this seems like an easy fix because there’s a public API for this now.

The fix inside React package is just to import the internal `checkPropTypes` module directly.
The fix inside renderers is to use the public API (`React.checkPropTypes`).

### CODE REVIEW

**Comment by gaearon** [comment]
Hmm this seems like the wrong way to do it because of a circular dep. Easier to kill this module and do it differently I think.

**Comment by gaearon** [comment]
OK should be ready for review!

**Comment by acdlite** [comment]
We can remove `checkPropTypes` and import from `prop-types/checkPropTypes` instead.

**Comment by gaearon** [comment]
Means we technically start to depend on it?

**Comment by acdlite** [comment]
Yeah we can remove `ReactPropTypes` too. I started a PR to migrate all the tests accordingly but haven't finished yet, will wrap up post-15.5 (tomorrow).

**Comment by acdlite** [comment]
Although, because of flat bundles it would just be dev dependency, right?

**Comment by gaearon** [comment]
Currently cjs flat bundles still use `require` for external deps (so that they can be deduped).

**Comment by gaearon** [comment]
I'll get this in for now so I can continue work on RN flat bundling, happy to change later.
